### PR TITLE
Table selection stuck fix

### PR DESCRIPTION
**Pull request by AlexanderReznik**
Starting selection in front of the table didn't trigger adding event listeners for mouseMove and mouseUp. Fixing this.

Before:
Selection stuck before

After:
Selection stuck after

### CODE REVIEW

File: packages/lexical-table/src/LexicalTableObserver.ts [hunk 1]
```diff
@@ -72,6 +72,7 @@ export class TableObserver {
   editor: LexicalEditor;
   tableSelection: TableSelection | null;
   hasHijackedSelectionStyles: boolean;
+  inSelectionProgress: boolean;
```
Comment (ivailop7) [inline packages/lexical-table/src/LexicalTableObserver.ts:75]: does it make sense to be called 'isSelecting' maybe?

Comment (ivailop7) [reply packages/lexical-table/src/LexicalTableObserver.ts:75]: since there's a method called 'isHighlightingCells' to keep things consistent

Comment (AlexanderReznik) [reply packages/lexical-table/src/LexicalTableObserver.ts]: Yes, I didn't like the previous name either
https://github.com/facebook/lexical/pull/5807/commits/60215b1dd2426293403d86a97dcd10ae00e4aa87

**Review by ivailop7 — COMMENTED** [review]
LGTM

**Review by ivailop7 — APPROVED** [review]

**Pull request merged**